File: problems/2404-most-frequent-even-element/solution.py
class Solution:
    def mostFrequentEven(self, nums: List[int]) -> int:
        counts = {}
        for num in nums:
            if num % 2 == 0:
                counts[num] = counts.get(num, 0) + 1
        
        if not counts:
            return -1
        
        max_freq = 0
        most_frequent = -1
        
        for num, freq in counts.items():
            if freq > max_freq:
                max_freq = freq
                most_frequent = num
            elif freq == max_freq and num < most_frequent:
                most_frequent = num
        
        return most_frequent

File: problems/2404-most-frequent-even-element/solution.java
class Solution {
    public int mostFrequentEven(int[] nums) {
        int[] count = new int[100001];
        int maxFreq = 0;
        int mostFrequent = -1;

        for (int num : nums) {
            if (num % 2 == 0) {
                count[num]++;
                if (count[num] > maxFreq) {
                    maxFreq = count[num];
                    mostFrequent = num;
                } else if (count[num] == maxFreq && num < mostFrequent) {
                    mostFrequent = num;
                }
            }
        }

        return mostFrequent;
    }
}